core/src/test/java/org/apache/calcite/test/TypeCoercionTest.java

```diff
@@ -747,8 +755,10 @@ protected Fixture(RelDataTypeFactory typeFactory,
       smallintType = this.typeFactory.createSqlType(SqlTypeName.SMALLINT);
       nullableSmallintType = this.typeFactory.createTypeWithNullability(smallintType, true);
       intType = this.typeFactory.createSqlType(SqlTypeName.INTEGER);
+      intJavaType = this.typeFactory.createJavaType(Integer.class);
       nullableIntType = this.typeFactory.createTypeWithNullability(intType, true);
       bigintType = this.typeFactory.createSqlType(SqlTypeName.BIGINT);
+      bigintJavaType = this.typeFactory.createJavaType(Long.class);
```

Comment:
Why not use BigInteger.class?

Reply:
The long type may be a timestamp

Reply:
According to JavaToSqlTypeConversionRules#rules, the Java type corresponding to SqlTypeName.BIGINT  is long. So I use `Long`.